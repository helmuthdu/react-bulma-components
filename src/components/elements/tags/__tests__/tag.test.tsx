import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import { Tag, Tags } from '..';

describe('Tag component', () => {
  // @ts-ignore
  it.each([[Tag], [Tags]])('should render', Component => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should concat classname in props with classname', () => {
    const { asFragment } = render(<Tag className="other-class this-is-a-test">Tag name</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should use inline styles', () => {
    const { asFragment } = render(<Tag style={{ width: 250 }}>Tag name</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should be Large', () => {
    const { asFragment } = render(<Tag size="large">Tag Large</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have info color', () => {
    const { asFragment } = render(<Tag color="info">Tag Large</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have close button', () => {
    const { asFragment } = render(<Tag close>Tag Large</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have remove button', () => {
    const { asFragment } = render(<Tag remove>Very large text with some extras</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trim content', () => {
    const { asFragment } = render(<Tag ellipsis>Very large text with some extras</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onClick event on close button', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <Tag close onClick={onClick}>
        close
      </Tag>
    );
    fireEvent.click(getByTestId('tag-delete'));
    expect(onClick).toHaveBeenCalled();
  });

  it('should trigger onClick event on remove button', () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Tag remove onClick={onClick} />);
    fireEvent.click(getByTestId('tag'));
    expect(onClick).toHaveBeenCalled();
  });

  it('should be group tags', () => {
    const { asFragment } = render(
      <Tags gapless>
        <Tag>Tag 1</Tag>
        <Tag remove />
      </Tags>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it(`Should use use color`, () => {
    const { asFragment } = render(<Tag color="white">tag</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });
});
