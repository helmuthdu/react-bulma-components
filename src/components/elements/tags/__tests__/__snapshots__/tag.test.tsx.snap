// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component Should use use color 1`] = `
<DocumentFragment>
  <span
    class="tag is-white"
    data-testid="tag"
  >
    <span
      class=""
    >
      tag
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag component should be Large 1`] = `
<DocumentFragment>
  <span
    class="tag is-large"
    data-testid="tag"
  >
    <span
      class=""
    >
      Tag Large
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag component should be group tags 1`] = `
<DocumentFragment>
  <span
    class="tagGroup.tsx has-addons"
  >
    <span
      class="tag"
      data-testid="tag"
    >
      <span
        class=""
      >
        Tag 1
      </span>
    </span>
    <span
      class="tag is-delete"
      data-testid="tag"
    />
  </span>
</DocumentFragment>
`;

exports[`Tag component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <span
    class="tag other-class this-is-a-test"
    data-testid="tag"
  >
    <span
      class=""
    >
      Tag name
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag component should have close button 1`] = `
<DocumentFragment>
  <span
    class="tag"
    data-testid="tag"
  >
    <span
      class=""
    >
      Tag Large
    </span>
    <button
      class="delete is-small"
      data-testid="tag-delete"
    />
  </span>
</DocumentFragment>
`;

exports[`Tag component should have info color 1`] = `
<DocumentFragment>
  <span
    class="tag is-info"
    data-testid="tag"
  >
    <span
      class=""
    >
      Tag Large
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag component should have remove button 1`] = `
<DocumentFragment>
  <span
    class="tag is-delete"
    data-testid="tag"
  />
</DocumentFragment>
`;

exports[`Tag component should render 1`] = `
<DocumentFragment>
  <span
    class="tag"
    data-testid="tag"
  >
    <span
      class=""
    />
  </span>
</DocumentFragment>
`;

exports[`Tag component should render 2`] = `
<DocumentFragment>
  <span
    class="tagGroup.tsx"
  />
</DocumentFragment>
`;

exports[`Tag component should trim content 1`] = `
<DocumentFragment>
  <span
    class="tag"
    data-testid="tag"
  >
    <span
      class="has-ellipsis"
    >
      Very large text with some extras
    </span>
  </span>
</DocumentFragment>
`;

exports[`Tag component should use inline styles 1`] = `
<DocumentFragment>
  <span
    class="tag"
    data-testid="tag"
    style="width: 250px;"
  >
    <span
      class=""
    >
      Tag name
    </span>
  </span>
</DocumentFragment>
`;
